base/cvd/cuttlefish/host/commands/cvd/server_command/create.cpp

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "host/commands/cvd/server_command/create.h"
+
+#include <sys/types.h>
+
+#include <cstdlib>
+#include <iostream>
+#include <optional>
+#include <string>
+
+#include <android-base/logging.h>
+#include <android-base/parseint.h>
+#include <android-base/strings.h>
+
+#include "common/libs/utils/contains.h"
+#include "common/libs/utils/files.h"
+#include "common/libs/utils/flag_parser.h"
+#include "common/libs/utils/result.h"
+#include "cuttlefish/host/commands/cvd/cvd_server.pb.h"
+#include "cuttlefish/host/commands/cvd/selector/cvd_persistent_data.pb.h"
+#include "host/commands/cvd/command_sequence.h"
+#include "host/commands/cvd/common_utils.h"
+#include "host/commands/cvd/selector/creation_analyzer.h"
+#include "host/commands/cvd/selector/instance_database_types.h"
+#include "host/commands/cvd/selector/instance_group_record.h"
+#include "host/commands/cvd/selector/selector_constants.h"
+#include "host/commands/cvd/server_command/server_handler.h"
+#include "host/commands/cvd/server_command/utils.h"
+#include "host/commands/cvd/types.h"
+
+namespace cuttlefish {
+namespace {
+
+constexpr char kSummaryHelpText[] =
+    "Create a Cuttlefish virtual device or environment";
+
+constexpr char kDetailedHelpText[] =
+    R"""(
+Usage:
+cvd create [--product_path=PATH] [--host_path=PATH] [--[no]start] [START_ARGS]
+cvd create --config_file=PATH [--[no]start]
+
+Creates and starts a new cuttlefish instance group.
+
+--host_path=PATH     The path to the directory containing the Cuttlefish Host
```

Comment:
what do you think of `host_out=PATH` and `product_out=PATH`? the terms `host_out` and `product_out` might be self explanatory if you are familiar with Android development.